go-controller/pkg/ovn/gateway.go

```diff
@@ -46,72 +47,77 @@ func (ovn *Controller) getGatewayLoadBalancer(physicalGateway string, protocol k
 	return loadBalancer, nil
 }
 
-func (ovn *Controller) createGatewayVIPs(protocol kapi.Protocol, sourcePort int32, targetIPs []string, targetPort int32) error {
-	klog.V(5).Infof("Creating Gateway VIPs - %s, %d, [%v], %d", protocol, sourcePort, targetIPs, targetPort)
-
-	// Each gateway has a separate load-balancer for N/S traffic
-
-	physicalGateways, _, err := ovn.getOvnGateways()
+func (ovn *Controller) forEachGatewayLB(protocols []kapi.Protocol, doCallback func(string, string, kapi.Protocol) error) error {
+	gateways, _, err := ovn.getOvnGateways()
 	if err != nil {
 		return err
 	}
+	for _, gateway := range gateways {
+		for _, protocol := range protocols {
+			loadBalancer, err := ovn.getGatewayLoadBalancer(gateway, protocol)
+			if err != nil || loadBalancer == "" {
+				return fmt.Errorf("gateway %s does not have load_balancer (%v)", gateway, err)
+			}
+			if err := doCallback(gateway, loadBalancer, protocol); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
 
-	for _, physicalGateway := range physicalGateways {
-		loadBalancer, err := ovn.getGatewayLoadBalancer(physicalGateway, protocol)
+func (ovn *Controller) createGatewayNodePortVIPs(protocol kapi.Protocol, port, targetPort int32, ips []string) error {
```

Comment:
it seems like there are a lot of changes in this commit that have nothing to do with adding support for external gateway. If that is the case, can you split that out into another commit to refactor/rename service functions?

Reply:
I will see what I can do to re-work that. Indeed somethings were changed as I went along and can be split in a separate PR (commit `42893795e982e8890b844a6d2ee57f962b85b495` can for example), I need to check how to do that for other smaller things.

Reply:
I created a separate PR for the `endpoints.go` / NodePort. related changes. I will create a separate commit in this PR for function renaming